src/goto-checker/bmc_util.cpp

```diff
@@ -225,3 +225,53 @@ void slice(
                    << " remaining after simplification" << messaget::eom;
 }
 
+std::vector<irep_idt> update_properties_status_from_symex_target_equation(
+  propertiest &properties,
+  const symex_target_equationt &equation)
+{
+  std::vector<irep_idt> updated_properties;
+
+  for(const auto &step : equation.SSA_steps)
+  {
+    if(!step.is_assert())
+      continue;
+
+    irep_idt property_id = step.get_property_id();
+
+    if(property_id.empty())
+      continue;
+
+    // Don't set false properties; we wouldn't have traces for them.
+    const auto status = step.cond_expr.is_true() ? property_statust::PASS
+                                                 : property_statust::UNKNOWN;
+    if(properties
+         .emplace(
+           property_id, property_infot{step.source.pc, step.comment, status})
+         .second)
+    {
+      updated_properties.push_back(property_id);
+    }
+    else
+    {
+      property_infot &property_info = properties.at(property_id);
+      property_statust old_status = property_info.status;
+      property_info.status |= status;
+
+      if(property_info.status != old_status)
+        updated_properties.push_back(property_id);
+    }
+  }
+
+  for(auto &property_pair : properties)
+  {
+    if(property_pair.second.status == property_statust::NOT_CHECKED)
+    {
+      // This could be a NOT_CHECKED, NOT_REACHABLE or PASS,
+      // but the equation doesn't give us precise information.
+      property_pair.second.status = property_statust::PASS;
```

Comment:
I don't understand what is happening here. The comment describing the function says that each status is set to `PASS` or `FAIL` for conditions that are constant, but I don't see where we check that it is constant.